resources/codegen_inputs/morx.rs

```diff
@@ -0,0 +1,82 @@
+#![parse_module(read_fonts::tables::morx)]
+
+/// The [morx (Extended Glyph Metamorphosis)](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6morx.html) table.
+#[tag = "morx"]
+table Morx {
+    /// Version number of the extended glyph metamorphosis table (either 2 or 3).
+    version: u16,
+    /// Not used; set to 0.
+    #[skip_getter]
+    #[compile(0)]
+    unused: u16,
+    /// Number of metamorphosis chains contained in this table.
+    n_chains: u32,
+    #[count($n_chains)]
+    chains: VarLenArray<Chain<'a>>,
+}
+
+/// A chain in a `morx` table.
+table Chain {
+    /// The default specification for subtables.
+    default_flags: u32,
+    /// Total byte count, including this header; must be a multiple of 4.
+    chain_length: u32,
+    /// Number of feature subtable entries.
+    n_feature_entries: u32,
+    /// The number of subtables in the chain.
+    n_subtables: u32,
```

Comment:
these three should all be computed at compile time; the first one with like `#[compile(self.compute_chain_length())]` and the other two with `#[compile(array_len($thing)]`


(this is true of a bunch of things here and I appreciate this doesn't include write-fonts support so I'll stop pointing them out; but will note that often when we add write-fonts support we forget to update these annotations, and then need to break the API later)

Reply:
Added some compile annotations. In general, compiling these AAT tables is going to take some hand written code regardless because most of the arrays are unsized and we simply read to the end of the available data. I'm assuming we won't need to compile these at all but who knows?